staging/src/k8s.io/kubectl/pkg/util/templates/templater.go

```diff
@@ -172,6 +173,14 @@ func (t *templater) rootCmdName(c *cobra.Command) string {
 	return t.rootCmd(c).CommandPath()
 }
 
+func (t *templater) reverseParentsNames(c *cobra.Command) []string {
+	reverseParentsNames := []string{}
+	for i := len(t.parents(c)) - 1; i >= 0; i-- {
+		reverseParentsNames = append(reverseParentsNames, t.parents(c)[i].Name())
+	}
```

Comment:
Since `t.parents(c)` is a func call that has to build the result every time it is called, how about if you call it once outside the loop, like this?
```suggestion
	parents := t.parents(c)
	for i := len(parents) - 1; i >= 0; i-- {
		reverseParentsNames = append(reverseParentsNames, parents[i].Name())
	}
```

Reply:
It is a very appropriate suggestion, thank you.